org.apache.seata.common.exception.FrameworkException: Unknown exception occurred

### Check Ahead

- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

- [ ] I am willing to try to fix this bug myself.


### Ⅰ. Issue Description

https://github.com/apache/incubator-seata/actions/runs/27112448257/job/80035966970
```
Error:  Tests run: 14, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 11.26 s <<< FAILURE! - in io.seata.saga.engine.StateMachineTests
Error:  io.seata.saga.engine.StateMachineTests.testSimpleScriptTaskStateMachine  Time elapsed: 0.755 s  <<< ERROR!
org.apache.seata.common.exception.FrameworkException: Unknown exception occurred
	at io.seata.saga.engine.StateMachineTests.lambda$testSimpleScriptTaskStateMachine$0(StateMachineTests.java:139)
	at io.seata.saga.engine.StateMachineTests.testSimpleScriptTaskStateMachine(StateMachineTests.java:135)
Caused by: java.lang.ClassCastException: class org.codehaus.groovy.GroovyBugError cannot be cast to class java.lang.Exception (org.codehaus.groovy.GroovyBugError is in unnamed module of loader 'app'; java.lang.Exception is in module java.base of loader 'bootstrap')
	at io.seata.saga.engine.StateMachineTests.lambda$testSimpleScriptTaskStateMachine$0(StateMachineTests.java:139)
	at io.seata.saga.engine.StateMachineTests.testSimpleScriptTaskStateMachine(StateMachineTests.java:135)
```

### Ⅱ. Describe what happened

_No response_

### Ⅲ. Describe what you expected to happen

_No response_

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

_No response_

### Ⅴ. Anything else we need to know?

_No response_

### Ⅵ. Environment

_No response_

--- a/dependencies/pom.xml
+++ b/dependencies/pom.xml
@@ -212,6 +212,10 @@
                         <groupId>org.apache.tomcat.embed</groupId>
                         <artifactId>tomcat-embed-el</artifactId>
                     </exclusion>
+                    <exclusion>
+                        <groupId>org.codehaus.groovy</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
                 </exclusions>
                 <type>pom</type>
                 <scope>import</scope>

--- a/saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/ScriptTaskStateHandler.java
+++ b/saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/ScriptTaskStateHandler.java
@@ -147,9 +147,13 @@ public class ScriptTaskStateHandler implements StateHandler, InterceptableStateH
                     scriptType,
                     e);
 
-            ((HierarchicalProcessContext) context).setVariableLocally(DomainConstants.VAR_NAME_CURRENT_EXCEPTION, e);
+            Exception exceptionToStore = (e instanceof Exception)
+                    ? (Exception) e
+                    : new RuntimeException("Script execution failed: " + e.getMessage(), e);
+            ((HierarchicalProcessContext) context)
+                    .setVariableLocally(DomainConstants.VAR_NAME_CURRENT_EXCEPTION, exceptionToStore);
 
-            EngineUtils.handleException(context, state, e);
+            EngineUtils.handleException(context, state, exceptionToStore);
         }
     }
 

